sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -313,29 +316,34 @@ async def refresh(self, **kwargs) -> None:
                 need_refresh = False
                 updated_sentinel_keys = dict(self._refresh_on)
                 for (key, label), etag in self._refresh_on.items():
-                    updated_sentinel = await self._client.get_configuration_setting(
-                        key=key,
-                        label=label,
-                        etag=etag,
-                        match_condition=MatchConditions.IfModified,
-                        headers=_get_headers("Watch"),
-                        **kwargs
-                    )
-                    if updated_sentinel is not None:
-                        logging.debug(
-                            "Refresh all triggered by key: %s label %s.",
-                            key,
-                            label,
+                    try:
+                        updated_sentinel = await self._client.get_configuration_setting(
+                            key=key,
+                            label=label,
+                            etag=etag,
+                            match_condition=MatchConditions.IfModified,
+                            headers=_get_headers("Watch"),
+                            **kwargs
                         )
-                        need_refresh = True
-
-                        updated_sentinel_keys[(key, label)] = updated_sentinel.etag
+                        if updated_sentinel is not None:
+                            logging.debug("Refresh all triggered by key: %s label %s.", key, label)
+                            need_refresh = True
+
+                            updated_sentinel_keys[(key, label)] = updated_sentinel.etag
+                    except HttpResponseError as e:
+                        if e.status_code == 404:
+                            # If the sentinel is not found, it means the key/label was deleted, so we should refresh
```

Comment:
What if the sentinel key doesnt exist at all? This would trigger refreshAll on every refresh invocation.